_posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md

```diff
@@ -0,0 +1,112 @@
+
+    ```ballerina
+    import ballerina/io;
+    import ballerina/persist;
+    import rainier.store;
+
+    store:Client sClient = check new ();
+
+    public function main() returns error? {
+        store:EmployeeInsert employee1 = {
+            id: "EMP001",
+            firstName: "John",
+            lastName: "Doe",
+            gender: "Male",
+            birthDate: {year: 1987, month: 7, day: 23},
+            hireDate: {year: 2020, month: 10, day: 10}
+        };
+
+        // Create a new employee record.
+        string[] employeeIds = check sClient->/employees.post([employee1]);
+        io:println(string `Inserted employee ids: ${employeeIds.toString()}`);
+
+
+        // Get the employee record with the ID: EMP001.
+        Employee|error employee = sClient->/employee/EMP001;
+
+        // Update the employee record with the ID: EMP001.
+        store:Employee employee = check sClient->/employees/[empId].put(
+            {hireDate: {year: 2014, month: 5, day: 1}
+        });
+
+        // Delete the employee record with the ID: EMP001.
+        store:Employee|error deleted = sClient->/employee/EMP001.delete();
+
+        // Get all the employee records.
+        stream<store:Employee, error?> employee = sClient->/employee;
+    }
+    ```
+
+In addition to the `Bal persist` feature, Swan Lake Update 6 brings a range of notable additions and improvements as described below.
+
+- **Electronic Data Interchange (EDI) support:** Swan Lake Update 6 introduces EDI message processing support in Ballerina. This capability allows the conversion of EDI messages to Ballerina records and vice versa, leveraging a specified EDI schema. Furthermore, it enables the generation of Ballerina records for EDI schemas in handling EDI data.
+    
+    The example code below demonstrates how a simple EDI schema for sales order data can be converted to Ballerina records.
+
+    ```ballerina
+    import ballerina/io;
+    import edi_to_record.sorder;
+
+    public function main() returns error? {
+        string ediText = check io:fileReadString("resources/order.edi");
+        sorder:SimpleOrder simpleOrder = check sorder:fromEdiString(ediText);
+        io:println("Order Id: ", simpleOrder.header.orderId);
+
+        foreach sorder:Items_Type item in simpleOrder.items {
+            io:println("Item: ", item.item, ", Quantity: ", item.quantity);
+        }
+    }
+    ```
+
+- **OpenAPI enum support:** Ballerina Open API tool now supports OpenAPI enums in both Ballerina client and service generation. The tool will generate Ballerina union types to represent the counterpart OpenAPI enum types.
+
+- **GraphQL field interceptors and interceptor configurations:** Starting from this release, you can utilize GraphQL field interceptors to apply an interceptor specifically to a particular resolver (resource/remote) method. Also, the newly introduced interceptor configurations allow you to change the behavior of a GraphQL service interceptor.
```

Comment:
`Starting from this release` is not needed?

Reply:
Changed to `now`.